src/issuance/batch.rs

```diff
@@ -0,0 +1,36 @@
+use std::collections::HashSet;
+
+use super::{verify_issue_bundle, AssetBase, IssueBundle, Signed};
+
+/// Batch validation context for Issuance.
+///
+#[derive(Debug, Default)]
+pub struct BatchValidator {
+    bundles: Vec<(IssueBundle<Signed>, [u8; 32])>,
+}
+
+impl BatchValidator {
+    /// Constructs a new batch validation context.
+    pub fn new() -> Self {
+        BatchValidator { bundles: vec![] }
+    }
+
+    /// Adds bundle to the validator.
+    pub fn add_bundle(&mut self, bundle: &IssueBundle<Signed>, sighash: [u8; 32]) {
+        self.bundles.push((bundle.clone(), sighash))
+    }
+
+    /// Batch-validates the accumulated bundles.
+    ///
+    /// Returns `true` if every bundle added to the batch validator is valid, or `false`
+    /// if one or more are invalid.
+    pub fn validate(self) -> bool {
+        // FIXME: take/save finalization set from/to the global state
+        let finalized = HashSet::<AssetBase>::new();
```

Comment:
We need a way to pass the `finilized` HashSet by the caller. Please add a way to transfer the parameter.
We can call it with an empty Set while global state is not implemented.

Reply:
Based on our latest discussion, I've reverted the changes related to BatchValidators as they are not used in Zebra: I have removed the modifications I made to the Orchard batch validator and completely removed Issuance batch validator I created before.